eras/alonzo/formal-spec/mir.tex

```diff
@@ -0,0 +1,415 @@
+      (\var{irReserves},~\var{irTreasury},~\var{deltaReserves},~\var{deltaTreasury})\leteq\var{i_{rwd}}
+      \\
+      \var{coin}\leteq\fun{mirTarget}~\var{c}
+      &
+      \var{coin} \geq 0
+      &
+      (\wcard,~\var{reserves})\leteq\var{acnt}
+      \\
+      \var{available}\leteq\var{reserves}+\var{deltaReserves}+
+      \left(\sum\limits_{\wcard\mapsto\var{val}\in\var{irReserves}} val\right)
+      \\
+      \var{coin} \leq \var{available}
+      \\
+      \var{i_{rwd}'}\leteq(\var{irReserves},~\var{irTreasury},~\var{deltaReserves}-\var{coin},~\var{deltaTreasury}+\var{coin})
+    }
+    {
+      \begin{array}{r}
+        \var{slot} \\
+        \var{ptr} \\
+        \var{acnt}
+      \end{array}
+      \vdash
+      \left(
+      \begin{array}{r}
+        \var{rewards} \\
+        \var{delegations} \\
+        \var{ptrs} \\
+        \var{fGenDelegs} \\
+        \var{genDelegs} \\
+        \var{i_{rwd}}
+      \end{array}
+      \right)
+      \trans{deleg}{c}
+      \left(
+      \begin{array}{c}
+        \var{rewards} \\
+        \var{delegations} \\
+        \var{ptrs} \\
+        \var{fGenDelegs}\\
+        \var{genDelegs} \\
+        \varUpdate{\var{i_{rwd}'}} \\
+      \end{array}
+      \right)
+    }
+  \end{equation}
+  \\~\\
+  \begin{equation}\label{eq:deleg-mir-t-to-r}
+    \inference[Deleg-Mir-T]
+    {
+      \var{c}\in \DCertMir
+      &
+      \fun{mirTarget}~\var{c}\in \Coin
+      &
+      \fun{mirPot}~\var{c}=\TreasuryMIR
+      \\
+      slot < \fun{firstSlot}~((\epoch{slot}) + 1) - \fun{StabilityWindow}\\
+      (\var{irReserves},~\var{irTreasury},~\var{deltaReserves},~\var{deltaTreasury})\leteq\var{i_{rwd}}
+      \\
+      \var{coin}\leteq\fun{mirTarget}~\var{c}
+      &
+      \var{coin} \geq 0
+      &
+      (\var{treasury},~\wcard)\leteq\var{acnt}
+      \\
+      \var{available}\leteq\var{treasury}+\var{deltaTreasury}+
+      \left(\sum\limits_{\wcard\mapsto\var{val}\in\var{irTreasury}} val\right)
+      \\
+      \var{coin} \leq \var{available}
+      \\
+      \var{i_{rwd}'}\leteq(\var{irReserves},~\var{irTreasury},~\var{deltaReserves}+\var{coin},~\var{deltaTreasury}-\var{coin})
+    }
+    {
+      \begin{array}{r}
+        \var{slot} \\
+        \var{ptr} \\
+        \var{acnt}
+      \end{array}
+      \vdash
+      \left(
+      \begin{array}{r}
+        \var{rewards} \\
+        \var{delegations} \\
+        \var{ptrs} \\
+        \var{fGenDelegs} \\
+        \var{genDelegs} \\
+        \var{i_{rwd}}
+      \end{array}
+      \right)
+      \trans{deleg}{c}
+      \left(
+      \begin{array}{c}
+        \var{rewards} \\
+        \var{delegations} \\
+        \var{ptrs} \\
+        \var{fGenDelegs}\\
+        \var{genDelegs} \\
+        \varUpdate{\var{i_{rwd}'}} \\
+      \end{array}
+      \right)
+    }
+  \end{equation}
+  \caption{MIR transfer Inference Rule}
+  \label{fig:dcert-mir-trans}
+\end{figure}
+
+\begin{figure}[ht]
```

Comment:
Not sure if it's worth it, but we could get rid of the code duplication in MIR by having a single rule for it and defining an `acnt'` to be either of the two options depending on the condition.

Reply:
that's true, as I mention in the comment above, that is actually how the code works.

Reply:
I forgot to attempt this after the first review, I've done it now, see what you think. It was a bit hard to keep the horizontal space within the table, but I think it looks okay-ish. :)

Reply:
I should probably make these "reduce copy" changes to the shelley spec as well now...

Reply:
ok, I've changed the Shelley MIR transitions as well now. The compare nicely now.